app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -206,6 +200,20 @@ class WireActivityViewModel @Inject constructor(
         }
     }
 
+    private fun observeLogoutState() {
+        viewModelScope.launch(dispatchers.io()) {
+            currentSessionFlow.get().invoke()
+                .distinctUntilChanged()
+                .collectLatest {
```

Comment:
change: if we want to handle all invalid sessions that are emitted then we shouldn't use `collectLatest` but regular `collect` 😉